spec/value_objects/results_variant_spec.rb

```diff
@@ -12,4 +12,24 @@
       ))
     end
   end
+
+  describe '#to_date' do
+    subject { described_class.new(value) }
+
+    context 'when the variant is `never_spent`' do
+      let(:value) { 'never_spent' }
+
+      it 'considers an infinity date' do
+        expect(subject.to_date).to eq(Date::Infinity.new)
+      end
+    end
+
+    context 'when the variant is `indefinite`' do
+      let(:value) { 'indefinite' }
+
+      it 'considers a very far in the future date but not infinite' do
+        expect(subject.to_date).to eq(Date.new(2049, 1, 1))
```

Comment:
I think we should add a test that fails if "Today" is 1 Jan 2049, so that someone can update the date to a even further future

Reply:
Yep good point, kind of a smoke test. Will add it.